esp/esdllib/README.md

```diff
@@ -319,224 +328,317 @@ Some script operations will add additional elements to the context document.
+        <privatekey>...</privatekey>
+        ...
+      </EspProtocol>
+      <EspService name="WsMyService"> ... </EspService>
+      <EspBinding service="WsMyService" protocol="https" type="EsdlBinding" netAddress="0.0.0.0" port="0">
+        <Authenticate method="mysecmgr" >
+          <Location path="/" resource="EnableSOAPInterface" required="Read" description="Root access to service"/>
+          <Feature path="AllowMyServiceExampleMethod" resource="AllowMyServiceExampleMethod" required="" description="Controls access to the ExampleMethod of WsMyService"/>
+          <Setting path="PremiumUserLevel" resource="Premium User Level" description="Value 1-3 indicating allowed level of premium features"/>
+          ...
+        </Authenticate>
+      </EspBinding>
+      <Authentication method="secmgrPlugin"/>
+    </EspProcess>
+    ...
+  </Software>
+</Environment>
+```
 
-| Name      | Usage Notes |
-|-----------|-------------|
-| @name     | Interpreted as a scalar value so quoting is not required. XPath expressions are not supported. String identifier used in trace output. Use is optional, but strongly encouraged. |
-| @target   | XPath expression that evaluates to the target of an operation or destination of an entry points output document node. As an entry point attribute it has these additional characteristics: <br><br> 1. Special macros for the configured back end query `{$query}` and the ESP SOAP request `{$request}` are available to use. <br> 2. The XPath must exist at runtime or an exception will be thrown. A future release will ensure the path exists without throwing an exception. |
-| @value    | A scalar value- XPath expressions are not supported. |
-| @source   | XPath expression giving the location in the script context where data will be read from. As an attribute of an endpoint element it can contain two special macros: `{$query}` for the configured back end query and `{$request}` for the ESP SOAP request.
+This deployment style requires a running dali server for the ESP process to load its *service bundle configurations* and ESDL definitions from. Refer to the *Dynamic ESDL* document on the HPCCSytems [documentation page](https://hpccsystems.com/training/documentation/all) for details on how to load the ESDL definition and the *service bundle configuration* to the dali. The *service bundle configuration* used here is the one outlined in the section [Service Bundle Configuration Commonalities](#service-bundle-configuration-commonalities).
 
-## Conventions
+### Debugging
 
-### Namespaces
+The `Method/@traceLevel` attribute controls how much pre- and post- entry point debugging information is sent to the trace file. Valid values are 1-10 with 10 being the most verbose and the default being 1. To see the ESDL Script context in trace log output, set `Method/@traceLevel=10`. After exiting each entry point the ESP prints out the contents of the context with the label `Entire script context after transforms:`
```

Comment:
Updated with a more complete list of what is printed out, which is all from the script context. The trace operations could print this out, but it would be more hassle to add the line(s) in the right places in the script(s). I think this gives a quick and simple way to toggle dumping diagnostics that you'd probably only ever want during development because its so  verbose. Script trace operations can also be used during development for some more targeted output, I think it makes most sense now to leave this independent of the scripted tracing.